File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.py
class Solution:
    # https://www.youtube.com/watch?v=yJv_ltADGuA
    # https://www.youtube.com/watch?v=EMKDoj5Sshk
    def removeDuplicates(self, s: str) -> str:
        stack = []

        for i in range(len(s)):
            if len(stack) > 0 and stack[-1] == s[i]:
                stack.pop()
            else:
                stack.append(s[i])

        return "".join(stack)

File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.rs
impl Solution {
    pub fn remove_duplicates(s: String) -> String {
        let mut stack: Vec<char> = Vec::new();

        for ch in s.chars() {
            if let Some(top) = stack.last() {
                if *top == ch {
                    stack.pop();
                } else {
                    stack.push(ch);
                }
            } else {
                stack.push(ch);
            }
        }

        stack.into_iter().collect()
    }
}
